core/src/main/java/org/apache/iceberg/avro/InternalReader.java

```diff
@@ -205,7 +205,6 @@ public ValueReader<?> primitive(Pair<Integer, Type> partner, Schema primitive) {
         case STRING:
           return ValueReaders.strings();
         case FIXED:
```

Comment:
As per Types.java, fixed also should be written and read as plain bytebuffers. 

Refer the testcase in `TestInternalWriter`

Reply:
I agree with this change.

Another check that this is correct is that `ByteBuffer` is used as the representation used by `FixedLiteral`, which must match the type of internal objects.